Automattic/woocommerce-payments
# fix: pricing decimal formatting for tokenized cart

Fixes tokenized cart for https://github.com/Automattic/woocommerce-payments/issues/8277

#### Changes proposed in this Pull Request









The amount displayed for zero-decimal currencies is wrong when using the PRBs w/ tokenized cart.
This is because Stripe has a concept of "zero decimal currency". Usually, we pass the PRBs a value expressed in cents. But for zero-decimal-currencies (like the Japanese Yen), the amount should be in Yen (they don't have Yen-cents).

Fixing.

Before:
<img width="1215" alt="Screenshot 2024-06-21 at 6 39 33 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/08cd1a68-80a6-415b-ba94-dfb00d475872">

After:
<img width="1173" alt="Screenshot 2024-06-21 at 6 22 45 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/673c5a1c-6e75-411b-8f5f-c1ad78463b8d">

#### Testing instructions





- Enable the "Enable Cart-Token implementation for PRBs" flag in your dev tools (you'll need to update them)
- Ensure you have Google Pay/Apple Pay enabled in the merchant's settings
- Go to WooCommerce > Settings > General > Number of decimals, change the decimals from `2` to `0` (for example - you could also try `1`)
- Go to WooCommerce > Settings > General > Currency, select JPY
- Go to a product page
- Click on the GooglePay/ApplePay button
- The amounts displayed in the GooglePay/ApplePay dialog should be the same as the one displayed on the product page


-------------------


**Post merge**
